### PR TITLE
PEO-319 Implement Bundler Audit runner

### PR DESCRIPTION
**Pull request by rostikkkk2**
[PEO-319](https://garage.atlassian.net/browse/PEO-319)

### Description

Need to implement bundler-audit runner.

Bundler-audit has a few types of criticality.

| BA | I |
|---|----|
|high|high|
|medium|medium|
|low|low|
|everything else|low|

### Before submitting the merge request make sure the following are checked

- [x] Followed the style guidelines of this project
- [x] Performed a self-review of own code
- [x] Wrote the tests that prove that fix is effective/that feature works

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,7 +12,7 @@
 /inquisition.yml
 .rake_tasks~
 .DS_Store
-Gemfile.lock
+/Gemfile.lock
```
Comment (chubchenko) [inline .gitignore]: What is purpose of this change?

File: inquisition.gemspec [hunk 2]
```diff
@@ -24,11 +24,14 @@ Gem::Specification.new do |spec|
...
+  spec.add_development_dependency 'pry'
   spec.add_development_dependency 'reek', '~> 5.0'
   spec.add_development_dependency 'rspec'
   spec.add_development_dependency 'rubocop', '~> 0.72.0'
   spec.add_development_dependency 'rubocop-performance', '~> 1.4'
   spec.add_development_dependency 'rubocop-rspec', '~> 1.33'
   spec.add_development_dependency 'simplecov'
   spec.add_development_dependency 'simplecov-lcov'
+
+  spec.add_dependency 'bundler-audit'
```
Comment (chubchenko) [inline inquisition.gemspec]: What about version?

File: lib/inquisition.rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
+require 'pry'
```
Comment (chubchenko) [inline lib/inquisition.rb]: Are you sure that he should be required here?

File: lib/inquisition/bundler/audit/runner.rb [hunk 4]
```diff
@@ -1,9 +1,35 @@
...
+          @issues = []
+          ::Bundler::Audit::Scanner.new.scan { |error| issues << create_error(error) }
+          issues
+        end
+
+        def create_error(error)
+          Inquisition::Issue.new(
+            level: error.advisory.criticality || LOW_LEVEL,
+            line: '',
+            runner: RUNNER_NAME,
```
Comment (chubchenko) [inline lib/inquisition/bundler/audit/runner.rb]: `self` instead of `RUNNER_NAME`

File: lib/inquisition/bundler/audit/runner.rb [hunk 5]
```diff
@@ -1,9 +1,35 @@
+require 'bundler/audit/cli'
+require 'bundler/audit/scanner'
+
 module Inquisition
   module Bundler
     module Audit
       class Runner < ::Inquisition::Runner
+        RUNNER_NAME = 'Bundler-audit'.freeze
+        LOW_LEVEL = :low
```
Comment (chubchenko) [inline lib/inquisition/bundler/audit/runner.rb]: Why do we need this constant here? Not sure that it related to the runner

File: lib/inquisition/bundler/audit/runner.rb [hunk 6]
```diff
@@ -1,9 +1,35 @@
+require 'bundler/audit/cli'
```
Comment (chubchenko) [inline lib/inquisition/bundler/audit/runner.rb]: Not sure that we need to require `cli`

File: lib/inquisition/bundler/audit/runner.rb [hunk 7]
```diff
@@ -1,9 +1,35 @@
...
   module Bundler
     module Audit
       class Runner < ::Inquisition::Runner
+        RUNNER_NAME = 'Bundler-audit'.freeze
+        LOW_LEVEL = :low
+        attr_reader :issues
+
         def call
-          puts 'Inquisition::Bundler::Audit::Runner'
+          ::Bundler::Audit::CLI.start [:update]
```
Comment (chubchenko) [inline lib/inquisition/bundler/audit/runner.rb]: Maybe would be better to work with [`Database`](https://github.com/rubysec/bundler-audit/blob/master/lib/bundler/audit/database.rb) instead if `CLI`, what do you think?

File: spec/fixtures/bundle/unpatched_gems/Gemfile [hunk 8]
```diff
@@ -0,0 +1,3 @@
+source 'https://rubygems.org'
```
Comment (chubchenko) [inline spec/fixtures/bundle/unpatched_gems/Gemfile]: This file can be removed

File: spec/fixtures/bundle/unpatched_gems/Gemfile.lock [hunk 9]
```diff
@@ -0,0 +1,35 @@
+GEM
```
Comment (chubchenko) [inline spec/fixtures/bundle/unpatched_gems/Gemfile.lock]: Same

File: .rubocop.yml [hunk 10]
```diff
@@ -10,3 +10,9 @@ Metrics/LineLength:
 
 Style/FrozenStringLiteralComment:
   Enabled: false
+
+Metrics/BlockLength:
```
Comment (chubchenko) [inline .rubocop.yml:14]: Why do we need these changes?

File: spec/inquisition/bundler/audit/runner_spec.rb [hunk 11]
```diff
@@ -1,4 +1,42 @@
 RSpec.describe Inquisition::Bundler::Audit::Runner do
   describe '#call' do
+    context 'when update bundle success' do
+      before { allow(Bundler::Audit::Database).to receive(:update!).and_return(true) }
+
+      it 'update' do
+        expect { described_class.call }.to output(/Updated ruby-advisory-db/).to_stdout
```
Comment (chubchenko) [inline spec/inquisition/bundler/audit/runner_spec.rb]: Not sure that something should be outputted

File: spec/inquisition/bundler/audit/runner_spec.rb [hunk 12]
```diff
@@ -1,4 +1,42 @@
...
+    context 'when update bundle failed' do
+      before { allow(Bundler::Audit::Database).to receive(:update!).and_return(false) }
+
+      it 'prints failure message' do
+        expect do
+          begin described_class.call
+          rescue SystemExit
+            StandardError
+          end
+        end.to output(/Failed updating ruby-advisory-db!/).to_stdout
```
Comment (chubchenko) [inline spec/inquisition/bundler/audit/runner_spec.rb]: Same

File: spec/inquisition/bundler/audit/runner_spec.rb [hunk 13]
```diff
@@ -1,4 +1,42 @@
...
+      end
+    end
+
+    context 'when check errors with bundler-audit scanner' do
```
Comment (chubchenko) [inline spec/inquisition/bundler/audit/runner_spec.rb]: These tests are not enough

Comment (chubchenko) [reply spec/inquisition/bundler/audit/runner_spec.rb]: http://www.betterspecs.org/#data

**Review by chubchenko — CHANGES_REQUESTED** [review]

File: .rubocop.yml [hunk 14]
```diff
@@ -10,3 +10,22 @@ Metrics/LineLength:
...
+RSpec/ExampleLength:
+  Max: 6
+
+RSpec/MessageSpies:
+  Exclude:
+    - 'spec/inquisition/bundler/audit/runner_spec.rb'
+
+RSpec/MultipleExpectations:
+  Exclude:
+    - 'spec/**/*.rb'
```
Comment (chubchenko) [inline .rubocop.yml:27]: It would be better if this cop is turned on (one `it` - one `expect `).

File: .rubocop.yml [hunk 15]
```diff
@@ -10,3 +10,22 @@ Metrics/LineLength:
...
+
+RSpec/ExampleLength:
```
Comment (chubchenko) [inline .rubocop.yml:29]: Same (this cop should be turned on)

File: .rubocop.yml [hunk 16]
```diff
@@ -10,3 +10,22 @@ Metrics/LineLength:
...
   Enabled: false
+
+Metrics/BlockLength:
+  Exclude:
+    - 'spec/inquisition/bundler/audit/runner_spec.rb'
+
+RSpec/ExampleLength:
+  Max: 6
+
+RSpec/MessageSpies:
```
Comment (chubchenko) [inline .rubocop.yml:21]: Same

File: .rubocop.yml [hunk 17]
```diff
@@ -10,3 +10,22 @@ Metrics/LineLength:
 
 Style/FrozenStringLiteralComment:
   Enabled: false
+
+Metrics/BlockLength:
+  Exclude:
+    - 'spec/inquisition/bundler/audit/runner_spec.rb'
+
+RSpec/ExampleLength:
```
Comment (chubchenko) [inline .rubocop.yml:18]: Same

File: spec/inquisition/bundler/audit/runner_spec.rb [hunk 18]
```diff
@@ -1,4 +1,51 @@
...
+      end
+
+      before do
+        scanner_instance.instance_variable_set(:@lockfile, lockfile)
+        allow(Bundler::Audit::Scanner::UnpatchedGem).to receive(:new).and_return(error_instance)
+        allow(Bundler::Audit::Scanner).to receive(:new).and_return(scanner_instance)
+        allow(Bundler::Audit::Database).to receive(:update!).and_return(true)
+      end
+
+      it 'scanner.scan return errors' do
```
Comment (chubchenko) [inline spec/inquisition/bundler/audit/runner_spec.rb:31]: `returns issue` instead of `scanner.scan return errors`

**Review by chubchenko — CHANGES_REQUESTED** [review]

**Comment by chubchenko** [comment]
@rostikkkk2 your branch has conflicts, can you resolve them please, thanks

**Review by Svatok — APPROVED** [review]

**Review by polykoi — APPROVED** [review]

**Pull request merged**